src/Utils.js

```diff
@@ -189,7 +189,7 @@ export function selectorError(selector, type = '') {
   );
 }
 
-export const isCompoundSelector = /([a-z]\.[a-z]|[a-z]\[.*\]|[a-z]#[a-z])/i;
+export const isCompoundSelector = /^[\.#]?-?[_a-z]+[_a-z0-9-]*[\.\[#]/i;
```

Comment:
> Even after that I'm not really sure how to reason out what it's matching against. I wonder if we could write this as multiple different regexes, instead of trying to combine it into one case?

Agreed, that's the approach I took with https://github.com/airbnb/enzyme/pull/591 because using a single regex was just too confusing. Breaking this into multiple regexes seems reasonable, but at a certain point it feels like we'd be implementing our own parser and I wonder if it would be better to just put the work into making https://github.com/airbnb/enzyme/issues/534.